test/customValidators_test.js

```diff
@@ -150,11 +174,149 @@ describe('validateServerVariables()', function () {
       }
     }`;
     const parsedInput = JSON.parse(inputString);
-  
-    expect(() => validateServerVariables(parsedInput, inputString, input)).to.throw('Not all server variables are described with variable object');
+
+    expect(() =>
+      validateServerVariables(parsedInput, inputString, input)
+    ).to.throw('Not all server variables are described with variable object');
   });
-});
 
+  // server with a variable that has enum and an example match one of them
+  it('should successfully validate the server variables that has enum and an example match one of them', async function () {
+    const inputString = `{ 
+        "servers": {
+          "dummy": {
+            "url": "http://localhost:{port}",
+            "description": "The production API server",
+            "protocol": "secure-mqtt",
+            "variables": {
+              "port": {
+                "enum": ["8883", "8884"],
+                "default": "8883",
+                "examples" : "8883"
+              }
+            }
+          }
+        }
+     }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    expect(validateServerVariables(parsedInput, inputString, input)).to.equal(
+      true
+    );
+  });
+
+  // server with a variable that  has only default and example, no enum
+  it('should successfully validate the server variables has only default and example, no enum', async function () {
+    const inputString = `{ 
+      "servers": {
+        "dummy":
+        {
+          "url": "http://localhost:{port}",
+          "description": "The production API server",
+          "protocol": "secure-mqtt",
+          "variables": {
+            "port": {
+              "default": "8883",
+              "examples" : "8883"
+            }
+          }
+        }
+      }
+    }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    expect(validateServerVariables(parsedInput, inputString, input)).to.equal(
+      true
+    );
+  });
+
+  // server with a variable that has one example and it doesn't match any of provided enum
+  it('should throw error on the server variables has one example and it does not match any of provided enum', async function () {
+    const inputString = `{ 
+      "servers": {
+        "dummy":
+        {
+          "url": "http://localhost:{port}",
+          "description": "The production API server",
+          "protocol": "secure-mqtt",
+          "variables": {
+            "port": {
+              "enum": ["8883", "8884"],
+              "examples" : "8882"
+            }
+          }
+        }
+      }
+    }`;
+
+    const parsedInput = JSON.parse(inputString);
+
+    expect(() =>
+      validateServerVariables(parsedInput, inputString, input)
+    ).to.throw(
+      'Please check your server variables. The example value is not one of the values from the enum'
+    );
```

Comment:
```suggestion
try {
      validateServerVariables(parsedInput, inputString, input);
    } catch (e) {
      expect(e.title).to.equal('Please check your server variables. The example value is not one of the values from the enum');
    }
```

when evaluating errors better close it in the try-catch, otherwise expect won't really check anything